python/cuml/test/test_nearest_neighbors.py

```diff
@@ -23,36 +23,118 @@
 
 from sklearn.neighbors import NearestNeighbors as skKNN
 from sklearn.datasets.samples_generator import make_blobs
+import cudf
+import pandas as pd
+import numpy as np
+from cuml.test.utils import array_equal
+
+import scipy.stats as stats
+
+
+def unit_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.unit)
+
+
+def quality_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.quality)
+
+
+def stress_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.stress)
+
+
+def predict(neigh_ind, _y, n_neighbors):
+
+    neigh_ind = neigh_ind.astype(np.int32)
+
+    ypred, count = stats.mode(_y[neigh_ind], axis=1)
+    return ypred.ravel(), count.ravel() * 1.0 / n_neighbors
+
+
+@pytest.mark.parametrize("nrows", [500, 1000, 10000])
+@pytest.mark.parametrize("ncols", [100, 1000])
+@pytest.mark.parametrize("n_neighbors", [10, 50])
+@pytest.mark.parametrize("n_clusters", [2, 10])
+def test_neighborhood_predictions(nrows, ncols, n_neighbors, n_clusters):
+
+    X, y = make_blobs(n_samples=nrows, centers=n_clusters,
+                      n_features=ncols, random_state=0)
+
+    X = X.astype(np.float32)
+
+    knn_cu = cuKNN()
+    knn_cu.fit(X)
+    neigh_ind = knn_cu.kneighbors(X, n_neighbors=n_neighbors,
+                                  return_distance=False)
+
+    labels, probs = predict(neigh_ind, y, n_neighbors)
+
+    assert array_equal(labels, y)
+
+
+def test_return_dists():
+    n_samples = 50
+    n_feats = 50
+    k = 5
+
+    X, y = make_blobs(n_samples=n_samples,
+                      n_features=n_feats, random_state=0)
+
+    knn_cu = cuKNN()
+    knn_cu.fit(X)
+
+    ret = knn_cu.kneighbors(X, k, return_distance=False)
+    assert not isinstance(ret, tuple)
+    assert ret.shape == (n_samples, k)
+
+    ret = knn_cu.kneighbors(X, k, return_distance=True)
+    assert isinstance(ret, tuple)
+    assert len(ret) == 2
```

Comment:
maybe just grab the first distance pair and confirm that it's right.. the current test would pass even if the  order got confused or we returned none or something for distance

Reply:
I’ll add a check for the dtype, since it should always be the same.